kong/llm/drivers/cohere.lua

```diff
@@ -12,102 +12,67 @@ local table_new = require("table.new")
 local DRIVER_NAME = "cohere"
 --
 
-local transformers_to = {
-  ["llm/v1/chat"] = function(request_table, model)
-    request_table.model = model.name
+local function merge_fields(request_table, model)
+  request_table.temperature = request_table.temperature or model.options.temperature
+  request_table.max_tokens = request_table.max_tokens or model.options.max_tokens
+  request_table.truncate = request_table.truncate or "END"
+  request_table.return_likelihoods = request_table.return_likelihoods or "NONE"
+  request_table.p = request_table.top_p or model.options.top_p
+  request_table.k = request_table.top_k or model.options.top_k
```

Comment:
Is it OK to have `request_table.p` be `false` if `top_p` is neither set in the request nor in the options?  In the other drivers, there code to merge the request table with the model options, but it is done in a per-option fashion.  It might be better to have a generalized merging function that works for all drivers, in the same way.

Reply:
https://github.com/Kong/kong/pull/12903/files#r1575191319